src/utils/plugin-metrics.ts

```diff
@@ -0,0 +1,74 @@
+import { IllegalOperationError } from '../utils/utils'
+import { Hub, MetricMathOperations, PluginConfig, PluginConfigId } from './../types'
+import { createPosthog } from './../worker/vm/extensions/posthog'
+
+interface PluginMetrics {
+    pluginConfig: PluginConfig
+    metrics: Record<string, number>
+}
+
+interface UpdateMetricPayload {
+    pluginConfig: PluginConfig
+    metricOperation: MetricMathOperations
+    metricName: string
+    value: number
+}
+
+export class PluginMetricsManager {
+    metricsPerPlugin: Record<PluginConfigId, PluginMetrics>
+
+    constructor() {
+        this.metricsPerPlugin = {}
+    }
+
+    sendPluginMetrics(hub: Hub): void {
+        for (const plugin of Object.values(this.metricsPerPlugin)) {
```

Comment:
Small nit: calling this `plugin` is a bit confusing, when it's a `PluginMetrics` object